### PR TITLE
Implement errors for listing operations.

### LINKED ISSUE #335: Add warning and safety mechanism to prevent unintentional list [buckets | keys] operations
Add warning and safety mechanism to prevent unintentional list [buckets | keys] operations.

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #335 (CLIENTS-1035) (CLIENTS-1035) (CLIENTS-1035) and [`CLIENTS-1035`](https://bashoeng.atlassian.net/browse/CLIENTS-1035)

### CODE REVIEW

File: src/RiakClient/RiakClient.cs [hunk 1]
```diff
@@ -659,7 +684,10 @@ public T Batch<T>(Func<IRiakBatchClient, T> batchFunction)
             {
                 try
                 {
-                    funResult = batchFunction(new RiakClient(conn));
+                    // TODO FIXME
```
Comment (alexmoore) [inline src/RiakClient/RiakClient.cs:687]: Is this a "fix now" or "fix later" kind of TODO?

Comment (lukebakken) [reply src/RiakClient/RiakClient.cs:687]: Later because it has never worked correctly.

File: src/RiakClient/Config/IRiakClusterConfiguration.cs [hunk 2]
```diff
@@ -53,6 +53,11 @@ public interface IRiakClusterConfiguration
         bool UseTtbEncoding { get; set; }
 
         /// <summary>
+        /// Set to <b>true</b> to disable exceptions thrown to prevent expensive operations.
+        /// </summary>
+        bool DisableListExceptions { get; set; }
```
Comment (alexmoore) [inline src/RiakClient/Config/IRiakClusterConfiguration.cs:58]: What do you think about renaming this `allowListing` or `allowListingOperations` like the other clients?  "DisableListExceptions" keeps throwing me off when seeing the negative `disable` and a boolean together.

**Comment by lukebakken** [comment]
The Python and Ruby clients use global settings called `disable_list_exceptions`. I figure that the .NET client should use that until it becomes `Command` based. For clients that use commands and builders, I'm using the `allow` form.

**Review by alexmoore — APPROVED** [review]
+1.

**Pull request merged**